File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.rb
def solution(code)
    answer = ''
    mode = false
    for i in 0..code.length-1
        if mode
            if code[i]=="1"
                mode=false
            else
                if i%2==1
                    answer=answer+code[i]
                end                
            end
        else
            if code[i]=="1"
                mode=true
            else
                if i%2==0
                    answer=answer+code[i]
                end
            end
        end
    end
    return answer.empty? ? "EMPTY":answer
end

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        StringBuilder answer = new StringBuilder();
        boolean mode = false;
        for(int i=0;i<code.length();i++){
            char k = code.charAt(i);
            if (mode){
                if(k!='1' && i%2==1)
                    answer.append(k);
                else if(k=='1')
                    mode=false;
            }
            else{
                if (k!='1' && i%2==0)
                    answer.append(k);
                else if (k=='1')
                    mode=true;
            }
                    
        }
        
        if (answer.toString().length()==0)
            answer.append("EMPTY");
        
        return answer.toString();
    }
}